smack-integration-test/src/main/java/org/jivesoftware/smackx/muc/MultiUserChatIntegrationTest.java

```diff
@@ -112,24 +157,517 @@ public void processMessage(Message message) {
+
+        mucAsSeenByTwo.addUserStatusListener(new UserStatusListener() {
+            @Override
+            public void adminRevoked() {
+                resultSyncPoint.signal("done");
+            }
+        });
+
+        createMUC(mucAsSeenByOne, "one-" + randomString);
+
+        final Resourcepart nicknameTwo = Resourcepart.from("two-" + randomString);
+        mucAsSeenByTwo.join(nicknameTwo);
+
+        mucAsSeenByOne.grantAdmin(conTwo.getUser().asBareJid());
+        mucAsSeenByOne.revokeAdmin(conTwo.getUser().asEntityBareJid());
+        try {
+            resultSyncPoint.waitForResult(timeout);
+        } catch (TimeoutException e) {
+            throw new AssertionError("Failed to receive presence");
+        } finally {
+            tryDestroy(mucAsSeenByOne);
+        }
+    }
+
+    /**
+     * Asserts that a user who is present when another user undergoes a role change receives that change as a presence update
+     *
+     * <p>From XEP-0045 § 5.2.2:</p>
+     * <blockquote>
+     * ...a MUC service implementation MUST change the user's affiliation to reflect the change and communicate that to
+     * all occupants...
+     * </blockquote>
+     *
+     * <p>From XEP-0045 § 10.7:</p>
+     * <blockquote>
+     * If the user is in the room, the service MUST then send updated presence from this individual to all occupants,
+     * indicating the loss of admin status by sending a presence element...
+     * </blockquote>
+     *
+     * @throws Exception when errors occur
+     */
+    @SmackIntegrationTest
+    public void mucAffiliationTestForWitnessingAdminRemoval() throws Exception {
+        EntityBareJid mucAddress = getRandomRoom("smack-inttest");
+
+        MultiUserChat mucAsSeenByOne = mucManagerOne.getMultiUserChat(mucAddress);
+        MultiUserChat mucAsSeenByTwo = mucManagerTwo.getMultiUserChat(mucAddress);
+        MultiUserChat mucAsSeenByThree = mucManagerThree.getMultiUserChat(mucAddress);
+
+        final ResultSyncPoint<String, Exception> resultSyncPoint = new ResultSyncPoint<>();
+
+        mucAsSeenByThree.addParticipantStatusListener(new ParticipantStatusListener() {
+            @Override
+            public void adminRevoked(EntityFullJid participant) {
+                resultSyncPoint.signal("done");
+            }
+        });
+
+        createMUC(mucAsSeenByOne, "one-" + randomString);
+        final Resourcepart nicknameTwo = Resourcepart.from("two-" + randomString);
+        final Resourcepart nicknameThree = Resourcepart.from("three-" + randomString);
+        mucAsSeenByTwo.join(nicknameTwo);
+        mucAsSeenByThree.join(nicknameThree);
+        mucAsSeenByOne.grantAdmin(conTwo.getUser().asBareJid());
+
+        mucAsSeenByOne.revokeAdmin(conTwo.getUser().asEntityBareJid());
+        try {
+            resultSyncPoint.waitForResult(timeout);
+        } catch (TimeoutException e) {
+            throw new AssertionError("Failed to receive presence");
```

Comment:
Always wrap the causing exception.

Come to think of it. The exception that is thrown because of the timeout should mark the test as failed and not abort the whole sinttest run. I am not sure if this is the case for `AssertionError`. In any way, why don't you simply *not* catch the `TimeoutException` and let it throw (I'd expect this exception type to mark the test as failed).

Reply:
I've removed the catch in hopes that it throws, although there's still a `finally` there too (which if I understand correctly means the cleanup will happen before the test bombs).

The AssertionError was because of this:
https://github.com/igniterealtime/Smack/blob/1df0763f92ce1775a1fb508701c3a9e5b91d2a1c/smack-integration-test/src/main/java/org/igniterealtime/smack/inttest/SmackIntegrationTestFramework.java#L496-L500
Although on reading through this time, I'm not sure that it has any deep effect.